File: 917-boats-to-save-people/boats-to-save-people.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        people.sort()

        l = 0 
        r = len(people) - 1
        res = 0

        while l <= r:

            if l < r and people[r] + people[l] <= limit:
                res += 1
                r -= 1
                l += 1
            
            else:
                r -= 1
                res += 1
                
        return res



        

File: 917-boats-to-save-people/boats-to-save-people.java
import java.util.*;

class Solution {
    public int numRescueBoats(int[] people, int limit) {

        Arrays.sort(people);

        int l = 0;
        int r = people.length - 1;
        int res = 0;

        while (l <= r) {

            if (l < r && people[l] + people[r] <= limit) {
                l++;
                r--;
            } else {
                r--;
            }

            res++;
        }

        return res;
    }
}